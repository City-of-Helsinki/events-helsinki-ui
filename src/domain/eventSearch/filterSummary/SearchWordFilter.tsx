import { IconCrossCircle } from 'hds-react';
import React from 'react';
import { useTranslation } from 'react-i18next';

import { FilterType } from '../../../common/components/filterButton/FilterButton';
import styles from './searchWordFilter.module.scss';

interface Props {
  onRemove: (value: string, type: FilterType) => void;
  text: string;
}

const SearchWordFilter: React.FC<Props> = ({ onRemove, text }) => {
  const { t } = useTranslation();

  const handleRemove = () => {
    onRemove(text, 'searchWord');
  };

  return (
    <div className={styles.searchWordFilter}>
      <div>‘{text}’</div>
      <button
        aria-label={t('commons.filter.ariaButtonRemove', {
          filter: text,
        })}
        onClick={handleRemove}
      >
        <IconCrossCircle />
      </button>
    </div>
  );
};

export default SearchWordFilter;
