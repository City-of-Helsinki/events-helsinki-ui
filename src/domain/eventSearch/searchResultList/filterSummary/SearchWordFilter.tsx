import React from "react";
import { useTranslation } from "react-i18next";

import IconDelete from "../../../../icons/IconDelete";
import styles from "./searchWordFilter.module.scss";

interface Props {
  onRemove: () => void;
  searchWord: string;
}

const SearchWordFilter: React.FC<Props> = ({ onRemove, searchWord }) => {
  const { t } = useTranslation();
  return (
    <div className={styles.searchWordFilter}>
      <div>‘{searchWord}’</div>
      <button
        aria-label={t("eventSearch.filters.buttonRemoveSearchWord")}
        onClick={onRemove}
      >
        <IconDelete />
      </button>
    </div>
  );
};

export default SearchWordFilter;
