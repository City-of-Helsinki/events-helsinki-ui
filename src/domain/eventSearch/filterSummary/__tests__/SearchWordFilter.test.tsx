import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';

import SearchWordFilter from '../SearchWordFilter';

const props = {
  onRemove: jest.fn(),
  text: 'text',
};

it('matches snapshot', () => {
  const { container } = render(<SearchWordFilter {...props} />);

  expect(container.firstChild).toMatchSnapshot();
});

it('calls onRemove callback when remove button is clicked ', () => {
  const onClickMock = jest.fn();
  render(<SearchWordFilter {...props} onRemove={onClickMock} />);

  expect(screen.queryByText(`‘${props.text}’`)).toBeInTheDocument();

  userEvent.click(screen.getByRole('button'));

  expect(onClickMock).toHaveBeenCalled();
  expect(onClickMock).toHaveBeenCalledWith(props.text, 'searchWord');
});
