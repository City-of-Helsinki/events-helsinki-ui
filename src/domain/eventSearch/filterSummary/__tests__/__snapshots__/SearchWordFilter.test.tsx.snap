// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  class="searchWordFilter"
>
  <div>
    ‘
    text
    ’
  </div>
  <button
    aria-label="eventSearch.filters.buttonRemoveSearchWord"
  >
    <svg
      class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        fill-rule="evenodd"
      >
        <path
          d="M0 0h24v24H0z"
        />
        <path
          d="M12 2c5.523 0 10 4.477 10 10s-4.477 10-10 10S2 17.523 2 12 6.477 2 12 2zm0 2a8 8 0 100 16 8 8 0 000-16zm2.828 3.757l1.415 1.415L13.413 12l2.83 2.828-1.415 1.415L12 13.413l-2.828 2.83-1.415-1.415L10.586 12 7.757 9.172l1.415-1.415L12 10.586l2.828-2.829z"
          fill="currentColor"
        />
      </g>
    </svg>
  </button>
</div>
`;
